public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx

```diff
@@ -247,7 +300,13 @@ export const AlertRuleForm: FC<Props> = ({ existing, prefill }) => {
           <CustomScrollbar autoHeightMin="100%" hideHorizontalTrack={true}>
             <div className={styles.contentInner}>
               <AlertRuleNameInput />
-              <QueryAndExpressionsStep editingExistingRule={!!existing} onDataChange={checkAlertCondition} />
+              <QueryAndExpressionsStep
+                editingExistingRule={!!existing}
+                prefill={!!prefill}
```

Comment:
I'm not sure if I understand why these props have to be passed all the way down in to the `QueryAndExpressionsStep`, it looks like we're using them to make some assertions on when to dispatch `setDataQueries`?

Reply:
We need to pass all the way down because:
- we need default queries for initialising the form with the default values. (when creating a new alert rule)
- we need also these default queries as you said in the `QueryAndExpressionsStep` to set queries once this async value is set, and also for the `onAddNewQuery` (to set the default)
- at the same time we need to keep updated this async default value in the `ExpressionEditor` when we are creating a new alert.

Reply:
Gotcha, that makes sense!